src/editData/editDataWebViewController.ts

```diff
@@ -0,0 +1,259 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import * as ed from "../sharedInterfaces/editData";
+
+import { ReactWebviewPanelController } from "../controllers/reactWebviewPanelController";
+import {
+    EditDataReducers,
+    EditDataWebViewState,
+} from "../sharedInterfaces/editData";
+import ConnectionManager from "../controllers/connectionManager";
+import { TreeNodeInfo } from "../objectExplorer/treeNodeInfo";
+import { ObjectExplorerUtils } from "../objectExplorer/objectExplorerUtils";
+import { Deferred } from "../protocol";
+import { EditDataService } from "../services/editDataService";
+import { EditSessionReadyNotification } from "../models/contracts/editData";
+import { NotificationHandler } from "vscode-languageclient";
+
+export class EditDataWebViewController extends ReactWebviewPanelController<
+    EditDataWebViewState,
+    EditDataReducers
+> {
+    constructor(
+        context: vscode.ExtensionContext,
+        private node: TreeNodeInfo,
+        private readonly connectionManager: ConnectionManager,
+        private readonly editDataService: EditDataService,
+        data?: EditDataWebViewState,
+    ) {
+        super(
+            context,
+            "editData",
+            data ?? {
```

Comment:
This is the initial state for `EditDataWebViewState`, right?  Probably good to have this situated with the definition for EditDataWebViewState.